engine/src/juliabox/cloud/compute.py

```diff
@@ -169,7 +168,7 @@ def can_terminate(is_leader):
     def get_redirect_instance_id():
         if not Compute.SCALE:
             Compute.log_debug("cluster size is fixed")
-            available_nodes = JBoxInstanceProps.get_available_instances(Compute.get_install_id())
+            available_nodes = Compute.impl.get_available_nodes()
```

Comment:
- `get_available_nodes` should be listed as part of the plugin contract (document section above) since plugins have to implement that
    - `Compute` should have a static method `get_available_nodes` that calls `Compute.impl.get_available_nodes()`.
- maybe `get_available_nodes` should be renamed to `get_avalable_instances` for consistency?